hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieFileIndex.scala

```diff
@@ -111,6 +111,9 @@ case class HoodieFileIndex(spark: SparkSession,
    */
   @transient private lazy val functionalIndex = new FunctionalIndexSupport(spark, metadataConfig, metaClient)
 
+  private val enableHoodieExtension = "org.apache.spark.sql.hudi.HoodieSparkSessionExtension"
```

Comment:
As per https://spark.apache.org/docs/latest/api/java/org/apache/spark/sql/SparkSessionExtensions.html : ```The extensions can also be used by setting the Spark SQL configuration property spark.sql.extensions. Multiple extensions can be set using a comma-separated list```

Instead of equalsIgnoreCase, can you use contains() API